app/components/application_date_component.html.erb

```diff
@@ -0,0 +1,12 @@
+<%=
+  case type
+  when :submitted_at
+    submitted_at_date
+  when :respond_by
+    respond_by_date
+  when :edit_by
+    edit_by_date
+  when :days_remaining
+    formatted_days_remaining
+  end
+%>
```

Comment:
It looks like this component is not doing a lot,
it is like a delegator (apart for `formatted_days_remaining`),
I am wondering if simple helper would not be simpler.

Reply:
Yeah, we thought the same thing when we did it. Helpers and components fill in the same role. We thought we would try it out in this case to see how it works.

It's slightly DRYer than the equivalent helper implementation, which would be like:

```ruby
def submitted_at_date
  dates = ApplicationDates.new(@application_form)
  dates.submitted_at.strftime('%e %B %Y')
end
```

Which is also less explicit because it doesn't require you to pass in an `@application_form`. I like the clear enforcing of passing in arguments that you get with `ActionView::Component`.

Reply:
I'm leaning that way too, it feels like https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/561/files#r347861143 is telling us that this is not an ideal use case for components.

Reply:
Cool, will refactor to helpers.

Reply:
Updated!